### PR TITLE
Added Arq. extension that activates request context.

### PR DESCRIPTION
**Pull request by manovotn**

### CODE REVIEW

File: tcks/microprofile-context-propagation/src/main/java/io/quarkus/arquillian/ArquillianBeforeAfterEnricher.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * Activates request context before test runs and shuts it down afterwards
+ */
+public class ArquillianBeforeAfterEnricher {
+
+    private static final String ERROR_MSG = "Arc container is not running, cannot activate CDI contexts!";
+
+    public void on(@Observes(precedence = -100) org.jboss.arquillian.test.spi.event.suite.Before event) throws Throwable {
+        ArcContainer container = Arc.container();
```
Comment (mkouba) [inline tcks/microprofile-context-propagation/src/main/java/io/quarkus/arquillian/ArquillianBeforeAfterEnricher.java]: Container may be null as well ;-)

File: tcks/microprofile-context-propagation/src/main/java/io/quarkus/arquillian/QuarkusArquillianLifecycleExtension.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package io.quarkus.arquillian;
+
+import org.jboss.arquillian.core.spi.LoadableExtension;
+
+public class QuarkusArquillianLifecycleExtension implements LoadableExtension {
```
Comment (mkouba) [inline tcks/microprofile-context-propagation/src/main/java/io/quarkus/arquillian/QuarkusArquillianLifecycleExtension.java]: There is no need to introduce a next arq extension. Just register the observer in the `io.quarkus.arquillian.QuarkusExtension`.

Comment (manovotn) [reply tcks/microprofile-context-propagation/src/main/java/io/quarkus/arquillian/QuarkusArquillianLifecycleExtension.java]: I wanted to activate it only for MP CP TCKs, putting it into `QuarkusExtension` will result in it being called in all TCKs...

Comment (mkouba) [reply tcks/microprofile-context-propagation/src/main/java/io/quarkus/arquillian/QuarkusArquillianLifecycleExtension.java]: Ok, in that case we should use a different package name.

File: tcks/microprofile-context-propagation/src/main/java/io/quarkus/arquillian/ArquillianBeforeAfterEnricher.java (hunk 1 above)
Comment (manovotn) [reply tcks/microprofile-context-propagation/src/main/java/io/quarkus/arquillian/ArquillianBeforeAfterEnricher.java]: Theoretically yes, practically it is started with all Arq. tests so not having it active at this point means we have problem elsewhere.

**Pull request merged**